src/coreclr/vm/multicorejitplayer.cpp

```diff
@@ -1349,53 +1158,169 @@ HRESULT MulticoreJitProfilePlayer::PlayProfile()
 
     while ((SUCCEEDED(hr)) && (nSize > sizeof(unsigned)))
     {
-        unsigned data   = * (const unsigned *) pBuffer;
-        unsigned rcdLen = data & 0xFFFFFF;
-        unsigned rcdTyp = data >> 24;
+        unsigned data1 = * (const unsigned *) pBuffer;
+        unsigned rcdTyp = data1 >> RECORD_TYPE_OFFSET;
+        unsigned rcdLen = 0;
+        
+        if (rcdTyp == MULTICOREJIT_MODULE_RECORD_ID)
+        {
+            rcdLen = data1 & 0xFFFFFF;    
+        }
+        else if (rcdTyp == MULTICOREJIT_MODULEDEPENDENCY_RECORD_ID)
+        {
+            rcdLen = sizeof(unsigned);
+        }
+        else if (rcdTyp == MULTICOREJIT_METHOD_RECORD_ID)
+        {
+            rcdLen = 2 * sizeof(unsigned);
+        }
+        else if (rcdTyp == MULTICOREJIT_GENERICMETHOD_RECORD_ID)
+        {
+            unsigned signatureLength = * (const unsigned short *) (((const unsigned *) pBuffer) + 1);
```

Comment:
It seems to be the intention in this part to check if there is enough space in the remaining buffer before decoding. From the loop condition we know that there is a `DWORD` worth of space but this is reading past that, so should check the remaining buffer length before dereferencing.

Reply:
Fixed